pkg/client/build.go

```diff
@@ -332,7 +332,7 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		return errors.Wrapf(err, "getting builder architecture")
 	}
 
-	bldr, err := c.getBuilder(rawBuilderImage)
+	bldr, err := c.getBuilder(rawBuilderImage, opts.RunImage, imgRegistry)
```

Comment:
Do we need to assume the registry from the output image? Why not just pass `opts.RunImage` only (I think we already require the `--run-image` to be fully qualified)?